### PR TITLE
feature(partner_sdk): add example for max value in destination capabilities

### PR DESCRIPTION
**Pull request by fivetran-niketkhandelwal**
Closes: https://fivetran.height.app/T-699973

### CODE REVIEW

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java [hunk 1]
```diff
@@ -48,26 +49,32 @@ public void configurationForm(ConfigurationFormRequest request, StreamObserver<C
...
-                                DataTypeMappingEntry.newBuilder().setFivetranType(DataType.BINARY).setMapTo(DestinationType.newBuilder().setName("BLOB").setMapTo(DataType.BINARY).build()).build(),
-                                DataTypeMappingEntry.newBuilder().setFivetranType(DataType.XML).setMapTo(DestinationType.newBuilder().setName("XML").setMapTo(DataType.XML).build()).build(),
-                                DataTypeMappingEntry.newBuilder().setFivetranType(DataType.STRING).setMapTo(DestinationType.newBuilder().setName("VARCHAR").setMapTo(DataType.STRING).build()).build(),
-                                DataTypeMappingEntry.newBuilder().setFivetranType(DataType.JSON).setMapTo(DestinationType.newBuilder().setName("OBJECT").setMapTo(DataType.JSON).build()).build()))
+                                DataTypeMappingEntry.newBuilder().setFivetranType(DataType.SHORT).setMapTo(DestinationType.newBuilder().setName("SHORT_INTEGER").setMapTo(DataType.SHORT).setMaxValue(MaxValue.newBuilder().setNumericParam(Integer.MAX_VALUE)).build()).build(),
```
Comment (ediril) [inline examples/destination/java/src/main/java/destination/DestinationServiceImpl.java:64]: All these examples are showing one to one mapping. Can we make a couple of them show other types of mappings, like we could map XML to "not used" and Double to Float or something..

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java [hunk 2]
```diff
@@ -48,26 +49,32 @@ public void configurationForm(ConfigurationFormRequest request, StreamObserver<C
 
     @Override
     public void capabilities(CapabilitiesRequest request, StreamObserver<CapabilitiesResponse> responseObserver) {
+        final int million = 1_000_000;
```
Comment (ediril) [inline examples/destination/java/src/main/java/destination/DestinationServiceImpl.java:52]: This variable name is not very descriptive. We should create different ones for the different numeric max values we are setting, the variable name should explain what it is

**Review by ediril — CHANGES_REQUESTED** [review]

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java (hunk 2 above)
Comment (fivetran-niketkhandelwal) [reply examples/destination/java/src/main/java/destination/DestinationServiceImpl.java]: changed

File: examples/destination/java/src/main/java/destination/DestinationServiceImpl.java (hunk 1 above)
Comment (fivetran-niketkhandelwal) [reply examples/destination/java/src/main/java/destination/DestinationServiceImpl.java]: - made `XML` unsupported
- mapped `SHORT` and `INT` to destination `INTEGER`
- mapped `DECIMAL` AND `FLOAT` to destination `FLOAT`
- mapped `NAIVE_TIME`, `NAIVE_DATETIME`, and `UTC_DATETIME` to destination `DATETIME`

**Review by fivetran-abdulsalam — APPROVED** [review]

**Review by ediril — APPROVED** [review]

**Pull request merged**